### ISSUE TITLE
Strict requirement to use the "@nrwl/nx-cloud" task runner for DTE

### ISSUE THREAD

**Issue by minijus** [nrwl/nx#14714]
### Current Behavior

Trying to upgrade `@nrwl/nx-cloud` to the latest v15.0.3 introduces new issue:
```
  >  NX   Invalid Task Runner Configuration
 
    To use Distributed Task Execution, your default task runner configuration must
    use the "@nrwl/nx-cloud" task runner.
    
    This can be adjusted in "nx.json".
```

In workspace we wrap `@nrwl/nx-cloud` task runner to pass custom self-signed CA. This allows developers to use nx seamlessly on local dev environment without a need to set `NODE_EXTRA_CA_CERTS` manually. Gist for wrapper: https://gist.github.com/minijus/609c78f08c25550bd325176551214ec8 

### Expected Behavior

Either `@nrwl/nx-cloud` task runner could add support for `caFile` option or remove explicit check for `@nrwl/nx-cloud` when starting agent.

### Steps to Reproduce

1. Try using Nx Cloud DTE with wrapped task executor.

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 18.13.0
   OS   : darwin arm64
   npm  : 9.3.1
   
   nx : 15.6.3
   @nrwl/angular : 15.6.3
   @nrwl/cypress : 15.6.3
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.6.3
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.6.3
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.6.3
   @nrwl/js : 15.6.3
   @nrwl/linter : 15.6.3
   @nrwl/nest : 15.6.3
   @nrwl/next : Not Found
   @nrwl/node : 15.6.3
   @nrwl/nx-cloud : 15.0.3
   @nrwl/nx-plugin : 15.6.3
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/webpack : 15.6.3
   @nrwl/workspace : 15.6.3
   @nrwl/vite : Not Found
   typescript : 4.9.4
   ---------------------------------------
   Local workspace plugins:
   	 ****/nx-plugin
   ---------------------------------------
   Community plugins:
   	 @ngrx/effects: 15.2.1
   	 @ngrx/entity: 15.2.1
   	 @ngrx/eslint-plugin: 15.2.1
   	 @ngrx/router-store: 15.2.1
   	 @ngrx/schematics: 15.2.1
   	 @ngrx/store: 15.2.1
   	 @ngrx/store-devtools: 15.2.1
```

**Comment by minijus** [comment]
Are the any updates on this? Our workspace is stuck on `@nrwl/nx-cloud` version 15.0.2 due to change that introduced task runner check.

**Comment by minijus** [comment]
As a temporary workaround it is possible to replace `@nrwl/nx-cloud` value with custom task runner in `distributed-agent.impl.js`.

```bash
sed -i -e 's/@nrwl\/nx-cloud/\.\/tools\/nxcloud-tasks-runner/g' ./node_modules/@nrwl/nx-cloud/lib/core/runners/distributed-agent/distributed-agent.impl.js
```

**Comment by StalkAltan** [comment]
@minijus As of this past Friday, we are beta testing the ability for users to pass custom proxy configurations with the `hpagent` package currently, and would expect the solution to be released in the next week or so. Thank you for your patience here.

**Comment by minijus** [comment]
> @minijus As of this past Friday, we are beta testing the ability for users to pass custom proxy configurations with the `hpagent` package currently, and would expect the solution to be released in the next week or so. Thank you for your patience here.

Thanks for having a look at this issue! `hpagent` would work for setting custom caFile if proxy would be needed as well. I do not see possibility to use `hpagent` without proxy itself.

**Comment by StalkAltan** [comment]
This has been released in 16.0.0, you can find light docs on this at: https://github.com/nrwl/nx-cloud-helm/blob/main/PROXY-GUIDE.md#nxcloud-runner-proxy-issues

This will make its way into nx.dev at some point in the future, but is more relevant for enterprise clients, so it lives with that info at this time.

**Comment by minijus** [comment]
Thank you @StalkAltan !
Just verified it with latest v16.0.1 and it worked fine.
